user.go

```diff
@@ -33,6 +33,15 @@ type UserGroup struct {
 	Name string `json:"name,omitempty" structs:"name,omitempty"`
 }
 
+type userSearchParam struct {
+	name  string
+	value string
+}
+
+type userSearch []userSearchParam
```

Comment:
Not sure if this is worth an entirely new type.  I think the code would be more readable if you just used `[]userSearchParam`.  Renaming it hides the fact that it's a list.

Reply:
I started with the `struct` but ended up in just building the query. In this case I think it's a matter of style - I like to name types (coming from Scala world), and this one is unexported. Can change it, no problem - if necessary.

Reply:
Yeah, I think it's more readable if you just use the `[]userSearchParam` instead of hiding it behind another type.  If someone else uses these functions it'll be more obvious that a list is being returned.

Reply:
It is only for the **internal** use, and I'd like to restrict the type of function return so people won't be able to pass something very custom with the incompatible unexported type.

just for the sake of purity and type-safety :)

Reply:
I'll merge this but I disagree.  Whether it's use is internal or not, you've created a pretty interesting functionality that, hopefully, someone else will extend.  At  that point, they'll be using your internal types.  

I think the readability of the obvious slice is better than hiding it behind a type that doesn't really add anything but does hide the fact that you're actually returning (or requesting) a slice.